app/scripts/models/reliers/oauth.js

```diff
@@ -157,6 +157,12 @@ define(function (require, exports, module) {
       // https://github.com/mozilla/fxa-oauth-server/blob/master/docs/api.md#post-v1authorization
       this.importSearchParamsUsingSchema(
           SIGNIN_SIGNUP_QUERY_PARAM_SCHEMA, OAuthErrors);
+
+      // OAuth reliers are not allowed to specify both a client_id and a service
```

Comment:
An OAuth relier has to specify a `client_id`, but they could be jerks and also specify `service=sync`, which caused us to be in a bad state. It's not a problem in practice, because we trust the reliers, but we might as well *make sure* they don't do so in the future too.